synapse/logging/context.py

```diff
@@ -890,12 +885,14 @@ def defer_to_threadpool(reactor, threadpool, f, *args, **kwargs):
             "Calling defer_to_threadpool from sentinel context: metrics will be lost"
         )
         parent_context = None
+        name = ""
```

Comment:
We probably actually want to set this to `str(curr_context)` in both cases and that will cause it to use either the parent context's name or the string "sentinel" depending on what was returned. 🤷